service/internal/graph/receiver.go

```diff
@@ -46,32 +48,40 @@ func (n *receiverNode) buildComponent(ctx context.Context,
 	if telemetry.NewPipelineTelemetryGate.IsEnabled() {
 		set.TelemetrySettings = telemetry.WithAttributeSet(set.TelemetrySettings, *n.Set())
 	}
-	var err error
+	tb, err := metadata.NewTelemetryBuilder(set.TelemetrySettings)
+	if err != nil {
+		return err
+	}
+
 	switch n.pipelineType {
 	case pipeline.SignalTraces:
 		var consumers []consumer.Traces
 		for _, next := range nexts {
 			consumers = append(consumers, next.(consumer.Traces))
 		}
-		n.Component, err = builder.CreateTraces(ctx, set, fanoutconsumer.NewTraces(consumers))
+		obsConsumer := obsconsumer.NewTraces(fanoutconsumer.NewTraces(consumers), tb.ReceiverProducedItems)
+		n.Component, err = builder.CreateTraces(ctx, set, obsConsumer)
```

Comment:
Not a blocking comment:

```suggestion
		n.Component, err = builder.CreateTraces(ctx, set, obsconsumer.NewTraces(fanoutconsumer.NewTraces(consumers), tb.ReceiverProducedItems))
```